macros/cross_db_utils/last_day.sql

```diff
@@ -0,0 +1,37 @@
+/*
+This function has been tested with dateparts of month and quarters. Further
+testing is required to validate that it will work on other dateparts.
+*/
+
+{% macro last_day(date, datepart) %}
+  {{ adapter_macro('dbt_utils.last_day', date, datepart) }}
+{% endmacro %}
+
+
+{% macro default__last_day(date, datepart) -%}
+    cast(
+        {{dbt_utils.dateadd('day', '-1',
+        dbt_utils.dateadd(datepart, '1', dbt_utils.date_trunc(datepart, date))
+        )}}
+        as date)
+{%- endmacro %}
+
+
+{% macro postgres__last_day(date, datepart) -%}
+
+    {%- if datepart == 'quarter' -%}
```

Comment:
just curious -- what's the problem with using `quarter` on Postgres? Does that datepart not exist?

Reply:
yep! that's exactly it.